### PR TITLE
for #3146. add a warning message if pie chart has more than one data set

### LINKED ISSUE #3146: PieChart `highlightValue(x:y:dataSetIndex:)` should provide error for dataSetIndex > 1
The issue in #2862 should provide feedback in the console that PieCharts can only accept an `dataSetIndex == 0`

### PR DESCRIPTION
**Pull request by liuxuan30**
fix #3146. add a warning message if pie chart has more than one data set

I specifically call super because we actually did nothing in pie chart.

### CODE REVIEW

File: Source/Charts/Data/Implementations/Standard/PieChartData.swift [hunk 1]
```diff
@@ -23,6 +23,23 @@ open class PieChartData: ChartData
...
+        {
+            if super.dataSets.count > 1
+            {
+                Swift.print("Found multiple data sets while pie chart only allows one")
+            }
+            return super.dataSets
+        }
+        set
+        {
+            super.dataSets = newValue
```
[the remark this answers is not in the document]
Comment (liuxuan30) [reply Source/Charts/Data/Implementations/Standard/PieChartData.swift]: I think it's too strict to add an assert.

[the remark this answers is not in the document]
Comment (jjatie) [reply Source/Charts/Data/Implementations/Standard/PieChartData.swift]: But you break the Chart if you try to do that. We need to assert here to make it clear that this isn't allowed.

[the remark this answers is not in the document]
Comment (jjatie) [reply Source/Charts/Data/Implementations/Standard/PieChartData.swift]: In fact we only need an assert on the setter, and nothing on the getter.

[the remark this answers is not in the document]
Comment (jjatie) [reply Source/Charts/Data/Implementations/Standard/PieChartData.swift]: Another way at looking at it is: This is not a suggestion, it is a requirement. Really, PieChartData shouldn't contain an array, but for now it does. Their app won't work properly if they are trying to set multiple datasets in the PieChartData and we need to make it as clear as possible that this is the case.

[the remark this answers is not in the document]
Comment (liuxuan30) [reply Source/Charts/Data/Implementations/Standard/PieChartData.swift]: Updated

File: Source/Charts/Data/Implementations/Standard/PieChartData.swift [hunk 2]
```diff
@@ -23,6 +23,20 @@ open class PieChartData: ChartData
...
+    /// - returns: All DataSet objects this ChartData object holds.
+    @objc open override var dataSets: [IChartDataSet]
+    {
+        get
+        {
+            return super.dataSets
+        }
+        set
+        {
+            assert(newValue.count > 1, "Found multiple data sets while pie chart only allows one")
```
Comment (jjatie) [inline Source/Charts/Data/Implementations/Standard/PieChartData.swift]: Whoops! You want `assert(newValue.count <= 1, "Found multiple data sets while pie chart only allows one")`

Comment (liuxuan30) [reply Source/Charts/Data/Implementations/Standard/PieChartData.swift]: What was I writing..

**Comment by liuxuan30** [comment]
@jjatie I tried the unit test on my side. It seems we should put in getter. the setter sometimes uses _dataSet directly so it won't trigger the assert. e.g. `chart.data = PieChartData(dataSet: dataSet)`

and we cannot change _dataSet to dataSet as the setter will call notifyDataChanged(), potential side effects such as performance drop. Not remember did we improved this in 4.0

putting into getter make things slightly easier.

**Comment by liuxuan30** [comment]
we could release 3.1 now

**Pull request merged**